cfme/tests/cloud/test_keypairs.py

```diff
@@ -95,7 +94,7 @@ def test_keypair_add_and_remove_tag(openstack_provider, keypairs):
         * Remove tag from Keypair
         * Also delete it.
     """
-    tag = ('Department', 'Accounting')
+    keypairs = KeyPairCollection()
```

Comment:
**REQUIRED:** This needs to have an appliance passed into it, ALL Collections objects should. In fact looking at the code:

```python
class KeyPairCollection(NavigatableMixin):
    """ Collection object for the :py:class: `cfme.cloud.KeyPair`. """
    def __init__(self, appliance):
        self.appliance = appliance
```

Looks like this has already been updated so that an appliance is a must, you can add it as a required fixture to the test and it will pick up the global fixture.

Reply:
Done, replaced with keypairs fixture